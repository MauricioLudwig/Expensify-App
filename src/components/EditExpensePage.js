import React from 'react';

const EditExpensePage = (props) => {
    console.log(props);
    return (
        <div>
            This is edit expense component
        </div>
    );
};

export default EditExpensePage;