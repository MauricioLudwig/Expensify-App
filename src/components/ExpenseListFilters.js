import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate } from '../actions/filters';

const ExpenseListFilters = (props) => (
    <div>
        <input type="text" value={props.filters.text} onChange={(e) => {
            props.dispatch(setTextFilter(e.target.value));
        }} />
        <select
            value={props.filters.sortBy}
            onChange={(e) => {

                const value = e.target.value;
                switch (value) {
                    case 'date':
                        props.dispatch(sortByDate());
                        break;
                    case 'amount':
                        props.dispatch(sortByAmount());
                        break;
                }

            }}>
            <option value="date">Date</option>
            <option value="amount">Amount</option>
        </select>
    </div>
);

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(ExpenseListFilters);