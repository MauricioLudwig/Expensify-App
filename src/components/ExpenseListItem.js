import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
    <div>
        <p>{description}</p>
        <p>{amount}</p>
        <p>{createdAt}</p>
        <button onClick={() => {
            dispatch(removeExpense({ id }))
        }}>Remove</button>
    </div>
);

export default connect()(ExpenseListItem);