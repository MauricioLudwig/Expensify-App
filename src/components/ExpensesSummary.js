import React, { Component } from 'react';
import { connect } from 'react-redux';
import getAmountSum from '../selectors/expenses-total';
import filterExpenses from '../selectors/expenses';
import numeral from 'numeral';

export class ExpensesSummary extends Component {

    render() {

        const { expenses } = this.props;
        const amountSum = numeral((getAmountSum(expenses) / 100)).format('$0,0.00');

        return (
            <div>
                {
                    expenses.length > 0
                    && <p>{`Viewing ${expenses.length} ${expenses.length === 1 ? 'expense' : 'expenses'} totalling ${amountSum}`}</p>
                }
            </div>
        );
    }

};

const mapStateToProps = state => ({
    expenses: filterExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);