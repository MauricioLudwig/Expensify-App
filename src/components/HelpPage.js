import React from 'react';

const HelpPage = () => (
    <div>
        This is help page component
    </div>
);

export default HelpPage;