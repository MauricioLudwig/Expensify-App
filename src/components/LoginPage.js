import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginPage extends Component {

    render() {
        return (
            <div>
                <button onClick={this.props.startLogin}>Login</button>
            </div>
        );
    }

};

const mapDispatchToProps = dispatch => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);