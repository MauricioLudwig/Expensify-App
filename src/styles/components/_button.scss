.button {
    background: $off-blue;
    color: white;
    padding: $m-size $l-size;
    font-size: $font-size-large;
    font-weight: 300;
    border: none;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}