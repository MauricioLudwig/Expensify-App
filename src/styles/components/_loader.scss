.loader {
    height: 100vh;
    width: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
}

.loader__image {
    height: 6rem;
    width: 6rem;
}