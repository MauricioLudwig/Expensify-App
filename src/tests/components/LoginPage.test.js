import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

let startLogin, wrapper;

beforeAll(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin} />);
});

test('add snapshot of LoginPage component', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});