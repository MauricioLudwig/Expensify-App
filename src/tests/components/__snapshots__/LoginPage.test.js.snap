// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add snapshot of LoginPage component 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
